File: C#/Basic logic/F1/F1_2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace F1
{
    internal class F1_2
    {
        static void Mainx(string[] args)
        {
            char znak;
            int odkaz, sub;

            Console.WriteLine("zadej znak");
            znak = char.Parse(Console.ReadLine());
            odkaz = asciF(znak); //funkce   
            asciS(znak, out sub); //subrutina

            Console.WriteLine(odkaz); //funkce
            Console.WriteLine(sub); //subrutina
        }
        static int asciF(char klavesa) //funkce
        {
            int vystup;
            vystup = klavesa;
            return vystup;
        }

        static void asciS(char klavesa, out int s) //subrutina
        {
            s = klavesa;
        }
    }
}

File: Java/Basics To Java/src/F1/F1_2.java
package F1;

import java.util.Scanner;

public class F1_2 {
	public static void mainx(String[] args) {
		Scanner sc = new Scanner(System.in);
		char znak;
		int odkaz;

		System.out.println("zadej znak");
		znak = sc.next().charAt(0);
		odkaz = asciF(znak); // funkce

		System.out.println(odkaz); // funkce
	}

	static int asciF(char klavesa) // funkce
	{
		int vystup;
		vystup = klavesa;
		return vystup;
	}
}
